filters/block/block_test.go

```diff
@@ -69,7 +81,145 @@ func TestBlock(t *testing.T) {
 
 		})
 	}
+}
+
+func TestMatcherErrorCases(t *testing.T) {
+	t.Run("maxBufferAbort", func(t *testing.T) {
+		r := &nonBlockingReader{initialContent: []byte("fppppppppp .class")}
+		toblockList := []toblockKeys{{str: []byte(".class")}}
+		bmb := newMatcher(r, toblockList, 5, maxBufferAbort)
+		p := make([]byte, len(r.initialContent))
+		_, err := bmb.Read(p)
+		if err != ErrMatcherBufferFull {
+			t.Errorf("Failed to get expected error %v, got: %v", ErrMatcherBufferFull, err)
+		}
+	})
+	t.Run("maxBuffer", func(t *testing.T) {
+		r := &nonBlockingReader{initialContent: []byte("fppppppppp .class")}
+		toblockList := []toblockKeys{{str: []byte(".class")}}
+		bmb := newMatcher(r, toblockList, 5, maxBufferBestEffort)
+		p := make([]byte, len(r.initialContent))
+		_, err := bmb.Read(p)
+		if err != nil {
+			t.Errorf("Failed to read: %v", err)
+		}
+	})
+
+	t.Run("maxBuffer read on closed reader", func(t *testing.T) {
+		pipeR, pipeW := io.Pipe()
+		initialContent := []byte("fppppppppp")
+		go pipeW.Write(initialContent)
+		toblockList := []toblockKeys{{str: []byte(".class")}}
+		bmb := newMatcher(pipeR, toblockList, 5, maxBufferBestEffort)
+		p := make([]byte, len(initialContent)+10)
+		pipeR.Close()
+		_, err := bmb.Read(p)
+		if err == nil || err.Error() != "io: read/write on closed pipe" {
+			t.Errorf("Failed to get correct read error: %v", err)
+		}
+	})
+
+	t.Run("maxBuffer read on initial closed reader", func(t *testing.T) {
+		pipeR, _ := io.Pipe()
+		initialContent := []byte("fppppppppp")
+		toblockList := []toblockKeys{{str: []byte(".class")}}
+		bmb := newMatcher(pipeR, toblockList, 5, maxBufferBestEffort)
+		p := make([]byte, len(initialContent)+10)
+		pipeR.Close()
+		bmb.Close()
+		_, err := bmb.Read(p)
+		if err == nil || err.Error() != "reader closed" {
```

Comment:
Error string matching is a good fit for `assert.EqualError` :grin:

Reply:
ah no thanks